R/computeAnnualDegreeDays.R

```diff
@@ -0,0 +1,191 @@
+#' of limit temperatures from raster data on (various) climate variables.
+#'
+#' For further processing, raster objects containing degree day data are written
+#' for an interval of ten years.
+#'
+#' @param ftas file name of data on near-surface atmospherical temperature
+#' @param tlim named list of limit temperature sequences for \code{HDD} and
+#' \code{CDD}
+#' @param countries SpatRaster defining (regional) aggregation boundaries
+#' @param pop SpatRaster containing population data
+#' @param factors data frame with degree day values for \code{temp/tlim}
+#' combination
+#' @param bait boolean, BAIT is used as ambient temperature
+#' @param frsds file name of data on surface downdwelling shortwave radiation
+#' (optional)
+#' @param fsfc file name of data on near-surface wind speed (optional)
+#' @param fhuss file name of data on near-surface specific humidity (optional)
+#' @param wBAIT named list containing BAIT weights (optional)
+#' @param baitPars SpatRaster containing regression parameters from
+#' \code{calcBAITpars} (optional)
+#'
+#' @return data frame containing regional population-weighted annual degree days
+#'
+#' @author Hagen Tockhorn
+#'
+#' @importFrom raster writeRaster
+#' @importFrom stringr str_split
+#' @importFrom terra writeCDF round
+#' @importFrom madrat readSource
+#' @importFrom dplyr mutate
+#' @importFrom magrittr %>%
+
+compStackHDDCDD <- function(ftas, tlim, countries, pop, factors, bait,
+                            frsds = NULL,
+                            fsfc  = NULL,
+                            fhuss = NULL,
+                            wBAIT = NULL,
+                            baitPars = NULL) {
+  # read cellular temperature
+  temp <- importData(subtype = ftas)
+
+  dates <- names(temp)
+
+  # optional: transform raw temperature into BAIT
+  if (bait) {
+    # note: easier to do in [C]
+    tempCelsius <- temp - 273.15   # [C]
+
+    # read and prepare bait input data
+    baitInput <- prepBaitInput(frsds, fsfc, fhuss) %>%
+      checkDates(tempCelsius)
+
+    # calculate bait
+    tempBAIT <- compBAIT(baitInput, tempCelsius, weight = wBAIT, params = baitPars)   # [C]
+
+    # convert back to [K]
+    temp <- tempBAIT + 273.15   # [K]
+  }
+
+  # round and assign dates
+  temp <- terra::round(temp, digits = 1)
+  names(temp) <- dates
+
+  # loop: type of degree day
+  hddcdd <- do.call(
+    "rbind", lapply(
+      c("HDD", "CDD"), function(typeDD) {
+        # loop: threshold temperatures
+        do.call(
+          "rbind", lapply(
+            tlim[[typeDD]], function(t) {
+              hddcddAgg <- compCellHDDCDD(temp, typeDD, t, factors)
+
+              hddcddAgg <- hddcddAgg %>%
+                aggCells(pop, countries) %>%
+                mutate("variable" = typeDD,
+                       "tlim"     = t)    # [C]
+
+              return(hddcddAgg)
+            }
+          )
+        )
+      }
+    )
+  )
```

Comment:
I think I exaggerated the line breaks and indentations back then. It's a matter of taste, but to me we could save some space here.
```suggestion
  hddcdd <- do.call(rbind, lapply(c("HDD", "CDD"), function(typeDD) {
    # loop: threshold temperatures
    do.call(rbind, lapply(tlim[[typeDD]], function(t) {
      compCellHDDCDD(temp, typeDD, t, factors) %>%
        aggCells(pop, countries) %>%
        mutate("variable" = typeDD,
               "tlim"     = t)    # [C]
    }))
  }))
```